src/hotspot/cpu/x86/x86.ad

```diff
@@ -1782,20 +1782,6 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
         return false; // Implementation limitation
       }
       break;
-    case Op_VectorCastB2X:
-    case Op_VectorCastS2X:
-    case Op_VectorCastI2X:
```

Comment:
I don't see any new code added for `VectorCastI2X`. Why is it so?

Reply:
This part is redundant, as above the only candidates for 256-bit vector on AVX1 is float and double. However, to cast to a 256-bit float vector we need a 256-bit int vector which is not supported, so a <= 128-bit int vector can only be casted to a 256-bit vector of type double.